inc/experiments/namespace.php

```diff
@@ -362,8 +372,19 @@ function register_post_ab_test( string $test_id, array $options ) {
 			'post',
 			'page',
 		],
+		'editor_scripts' => [],
 	] );
 
+	// Handle post type support filtering and registration.
+	$options['post_types'] = apply_filters( "altis.experiments.features.{ $test_id }.post_types", array_filter( (array) $options['post_types'] ) );
+	foreach ( $options['post_types'] as $post_type ) {
+		add_post_type_support( $post_type, "altis.experiments.{ $test_id }" );
+	}
```

Comment:
I think this reverses the logic I had. The results of `get_post_types_by_support( "altis.experiments.{ $test_id }" )` would need to be merged into `$options['post_types']` as that value is checked in other places like the background task handler.

Reply:
With the filter I wasn't really thinking of it as the core way of modifying a test at the time necessarily, I should've thought about that more and made a better implementation at the time. It's not strictly a filter that should be used anyway - that's up to us to decide, and I agree having both methods makes things confusing. I'm leaning towards getting rid of the filter just for the post types setting.

Previously if I'd used `add_post_type_support( 'type', 'altis.experiments.titles' )` it would have worked to add support for the titles test to that post type. That's what we're losing here at the moment with this change that I wanted to call out.

Ultimately the outcome I think we want is that it should be easy to add or remove support for a given test, and there should be a single canonical source of truth for getting supported post types, one of:

- `get_post_types_by_support( 'altis.experiments.<test>' )`
- `$post_ab_tests[ <test> ]['post_types']`

If we go with the post type support approach then we'll need to use `get_post_types_by_support( 'altis.experiments.<test>' )` as the way to get the array of post types. So we'd need to update any references to `$post_ab_tests[ <test> ]['post_types']`. What do you think?

Regarding filters in WP there is a pattern of having a filter or action that fires when registering something, like `register_post_type_args` and `registered_post_type` that might be a better pattern to follow than just having a filter for the post types option here, makes it much more pluggable.

Reply:
Yeah, I agree that using `get_post_types_by_support()` and getting rid of the filter makes the most sense, and about adding an action after registration. 👍